### ISSUE TITLE
Any error in a script stops all functionality such as jumping to declarations, context help, etc.

### ISSUE THREAD

**Issue by chucklepie** [godotengine/godot#44888]
**Godot version:**
3.2.3

**OS/device including version:**
Linux

**Issue description:**
I tried searching for a similar fault and didn't know if this might come under any of existing faults logged with improving error handling in the editor.

Anyway, if there are any errors in a script then you lose all ability to jump to declarations, context '.' disappears, etc.

Simple example below there's an error (it's just a tabbing error, but it's just for reference, there could be more) I want to CTRL+Click on a line above to jump to the declaration of 'help_smart_screen', but I can't, I have to fix the bug first.

![image](https://user-images.githubusercontent.com/12863685/103478481-f84f3e80-4dbe-11eb-81df-a3a00bb58c02.png)

An editor should not be stopping all basic editor functionality just because there's an error. No other IDE/Editor does this.

**Comment by volzhs** [comment]
seems related to #27424 #23264

**Comment by chucklepie** [comment]
@volzhs I don't think so, the second is about the order of context items in the dropdown and the first (which is mine also) is about an error when you try to use context at the end of a block of code. This bug is simply nothing works when there is a bug somewhere in the script.

**Comment by KoBeWi** [comment]
Any reproduction project? It seems to work fine with simple code:
```
func _ready() -> void:
	var node = Node.new()
	print(node.name)
	error
```
You can Ctrl+Click `Node` and `name` just fine.

**Comment by Omicron666** [comment]
i often get that error, can't reproduce that, right now, though.

**Comment by Calinou** [comment]
This has always been the case in Godot 3.2 and prior. It can't be solved until Godot 4.0.

**Comment by chucklepie** [comment]
Wohoo. Close the case and let's party 😉
